src/src/transport/uart.c

```diff
@@ -147,6 +147,8 @@ otError otPlatUartFlush(void)
     {
         // Wait until the transmission is done
     }
+    sTransmitBuffer = NULL;
```

Comment:
Hello, thank you for the PR and sorry for the late response. 
The only side effect of this proposal is that usually, after otPlatUartFlush() is called the otPlatUartSendDone() callback would be invoked during the next main loop run. After this change, when otPlatUartFlush() is called `sTransmitBuffer = NULL;` will make function `processTransmit()` exit immediately without setting `sTransmitDone   = false;`. Maybe instead of setting `sTransmitBuffer` to `NULL` here, we should just call `processTransmit()`? This way we would have a single point of cleanup. Moreover, it would call `otPlatUartSendDone()` properly. I know that CLI code expects that flushing without an error is an indicator of successful transmission, but I believe that such notification should come from the driver layer (every `otPlatUartSendDone()` ends with `otPlatUartSendDone()`, which would be omitted in the current proposal.

Reply:
SGTM. I've verified tthat his works.